modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java

```diff
@@ -661,31 +672,90 @@ private SystemView<?> createSystemViewColumnsView() {
+                .addColumn("REPLICAS", INT32, CatalogZoneDescriptor::replicas)
+                .addColumn("DATA_NODES_AUTO_ADJUST_SCALE_UP", INT32, CatalogZoneDescriptor::dataNodesAutoAdjustScaleUp)
+                .addColumn("DATA_NODES_AUTO_ADJUST_SCALE_DOWN", INT32, CatalogZoneDescriptor::dataNodesAutoAdjustScaleDown)
+                .addColumn("DATA_NODES_FILTER", STRING, CatalogZoneDescriptor::filter)
+                .addColumn("IS_DEFAULT_ZONE", BOOLEAN, isDefaultZone())
                 .dataProvider(SubscriptionUtils.fromIterable(() -> catalog(latestCatalogVersion()).zones().iterator()))
                 .build();
     }
 
+    private SystemView<?> createSystemViewIndexesView() {
+        return SystemViews.<CatalogIndexViewDescriptor>clusterViewBuilder()
+                .name("INDEXES")
+                .addColumn("INDEX_ID", INT32, CatalogIndexViewDescriptor::id)
+                .addColumn("INDEX_NAME", STRING, CatalogIndexViewDescriptor::name)
+                .addColumn("TABLE_ID", INT32, CatalogIndexViewDescriptor::tableId)
+                .addColumn("TABLE_NAME", STRING, CatalogIndexViewDescriptor::tableName)
+                .addColumn("SCHEMA_ID", INT32, CatalogIndexViewDescriptor::schemaId)
+                .addColumn("SCHEMA_NAME", STRING, CatalogIndexViewDescriptor::schemaName)
+                .addColumn("TYPE", STRING, CatalogIndexViewDescriptor::type)
+                .addColumn("UNIQUE", BOOLEAN, CatalogIndexViewDescriptor::unique)
+                .addColumn("COLUMNS", STRING, CatalogIndexViewDescriptor::columnsString)
+                .addColumn("STATUS", STRING, CatalogIndexViewDescriptor::status)
+                .dataProvider(getIndexesDataProvider())
+                .build();
+    }
+
+    private Publisher<CatalogIndexViewDescriptor> getIndexesDataProvider() {
+        return SubscriptionUtils.fromIterable(() ->
+                catalog(latestCatalogVersion())
+                        .indexes()
+                        .stream()
+                        .filter(it -> it.status() != STOPPING)
+                        .map(this::toCatalogIndexView)
+                        .iterator()
+        );
+    }
+
+    private CatalogIndexViewDescriptor toCatalogIndexView(CatalogIndexDescriptor index) {
+        Catalog catalog = catalog(latestCatalogVersion());
+        CatalogTableDescriptor table = catalog.table(index.tableId());
+
+        return new CatalogIndexViewDescriptor(
+                index.id(),
+                index.name(),
+                index.indexType().name(),
+                index.tableId(),
+                table.name(),
+                table.schemaId(),
+                catalog.schema(table.schemaId()).name(),
+                index.unique(),
+                getIndexColumns(index),
+                index.status().name()
+        );
+    }
+
+    private static String getIndexColumns(CatalogIndexDescriptor indexDescriptor) {
```

Comment:
Let's move this method elsewhere, like to `CatalogIndexViewDescriptor` as it seems pretty specific for this view logic